src/Domain/LeanCode.DomainModels.MassTransitRelay/Middleware/ConsumeMessageOnceFilter.cs

```diff
@@ -30,6 +30,7 @@ public ConsumeMessageOnceFilter(IConsumedMessagesContext consumedMessages)
             await consumedMessages.ConsumedMessages
                 .Where(m => m.ConsumerType == msg.ConsumerType && m.MessageId == msg.MessageId)
                 .AnyAsync()
+            && context.GetRedeliveryCount() == 0
```

Comment:
This means that we actually do not acknowledge consumption of any redelivered message, which might be unexpected for the user.